src/net.cpp

```diff
@@ -2176,12 +2176,19 @@ void CConnman::SetNetworkActive(bool active)
     uiInterface.NotifyNetworkActiveChanged(fNetworkActive);
 }
 
-CConnman::CConnman(uint64_t nSeed0In, uint64_t nSeed1In) : nSeed0(nSeed0In), nSeed1(nSeed1In)
+CConnman::CConnman(uint64_t nSeed0In, uint64_t nSeed1In)
+    : m_anchors_db_path{GetDataDir() / "anchors.dat"}, nSeed0(nSeed0In), nSeed1(nSeed1In)
 {
     SetTryNewOutboundPeer(false);
 
     Options connOptions;
     Init(connOptions);
+
+    m_anchors = ReadAnchors(m_anchors_db_path);
+    if (m_anchors.size() > m_max_outbound_block_relay) {
+        // Keep our policy safe.
+        m_anchors.clear();
```

Comment:
The worst case is assumed. If condition `m_anchors.size() > m_max_outbound_block_relay` is satisfied, we could suppose that the `anchors.dat` file is wrong or corrupted.

Reply:
No you're melting 2 different problems, tracking the anchors set and connecting to peers among this one. I don't understand why reaching the `m_max_outbound_block_relay` means we have to delete our whole repository of maybe-good anchors peers. 

In the scenario you're aiming, if victim restarts and the 2 anchors peers are honest but unreliable, you're going to drop them and pick maybe new outbound peers from attacker-provided ones. You can avoid this if you have a bigger set of _potential_ anchor peers.

Reply:
the current logic is minimal, which I think makes the most sense for this iteration because it keeps it simpler to reason about. 

with this implementation, its still possible for an attacker to poison the anchors and execute an eclipse attack, but increases the cost of such attack. 

until we develop logic around ensuring the anchors are not malicious, I don't think it makes sense to give priority to all historic anchors. I think doing so would sometimes help, but other times degrade security. eg. if you historically were connected to a malicious peer but now your addrman is clean.

Reply:
the logic implemented in this PR is minimal, and I think that makes sense for keeping it simple to reason about. 

with this implementation, its possible for an attacker to poison the anchors and execute an eclipse attack, its just increases the cost of doing so. 

until we develop more robust logic around ensuring our anchors are not malicious, I don't think it makes sense to give priority to all historic anchors. I think it could sometimes be beneficial but other times degrade security, eg. if you historically were connected to a malicious peer but now your addrman is clear.